### PR TITLE
Implemented RSA encryptor

### PR DESCRIPTION
**Pull request by pragyanmehrotra**
Hi, @rafaeelaudibert I took your advice and tried to code according to as the other encryptors were coded, the catch is the input text for decrypt should be in hex.

This has reference to Issue #19

### CODE REVIEW

File: encryptor/encryptors/rsa.py [hunk 1]
```diff
@@ -0,0 +1,85 @@
+from ..errors.not_allowed_value import NotAllowedValue
+from ..errors.wrong_type_parameter import WrongTypeParameter
+from binascii import hexlify,unhexlify
+
+
+class Rsa:
+
+	Default_E = 65537
```
Comment (rafaeelaudibert) [inline encryptor/encryptors/rsa.py]: Please, name these different default parameters in uppercase, i.e.: `DEFAULT_E`

File: encryptor/encryptors/rsa.py [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+	Default_P = 296364335885826735335603657657712346243
+	Default_Q = 261610179866133310725739923400935476237
+	Default_N = Default_P*Default_Q
+
+	@staticmethod
+	def encrypt(text: str, n: int = Default_N, e: int = Default_E):
+		if (n is None):
```
Comment (rafaeelaudibert) [inline encryptor/encryptors/rsa.py]: Remove the parenthesis around the condition

File: encryptor/encryptors/rsa.py [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+			n = Default_N
+		else:
+			try:
+				n = int(n)
+			except ValueError:
+				raise WrongTypeParameter("n", int, type(n))
+
+		if (e is None):
```
Comment (rafaeelaudibert) [inline encryptor/encryptors/rsa.py]: Remove the parenthesis around the condition

File: encryptor/encryptors/rsa.py [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+
+class Rsa:
+
+	Default_E = 65537
+	Default_P = 296364335885826735335603657657712346243
+	Default_Q = 261610179866133310725739923400935476237
+	Default_N = Default_P*Default_Q
+
+	@staticmethod
+	def encrypt(text: str, n: int = Default_N, e: int = Default_E):
```
Comment (rafaeelaudibert) [inline encryptor/encryptors/rsa.py]: You could use Python ternary operator in the if checks below, like the following snippet:

```python3
try:
  n = DEFAULT_N if n is None else int(n)
except ValueError:
  raise WrongTypeParameter("n", int, type(n))
```

File: encryptor/encryptors/rsa.py [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+
+		if (e is None):
+			e = Rsa.Default_E
+		else:
+			try:
+				e = int(e)
+			except ValueError:
+				raise WrongTypeParameter("e", int, type(e))
+
+		message = int(hexlify(bytes(text,'utf-8')),16)
```
Comment (rafaeelaudibert) [inline encryptor/encryptors/rsa.py]: Please add spaces between the function parameters, for all the called functions below

File: encryptor/encryptors/rsa.py [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+		encrypted_message = pow(message,e,n)
+		return hex(encrypted_message)[2:].split('L')[0]
+
+	@staticmethod
+	def decrypt(text: str, p: int = Default_P, q: int = Default_Q, e: int = Default_E):
```
Comment (rafaeelaudibert) [inline encryptor/encryptors/rsa.py]: Same thing with ternary operator as explained above

File: encryptor/encryptors/rsa.py [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+			return (b, 0, 1)
+		else:
+			g, y, x = Rsa.egcd(b % a, a)
+			return (g, x - (b//a)*y, y)
+
+	@staticmethod
+	def modinv(a, m):
+		g, x, y = Rsa.egcd(a, m)
+		if g != 1:
+			raise Exception("Modular Inverse does not exist")
```
Comment (rafaeelaudibert) [inline encryptor/encryptors/rsa.py]: You may consider creating a custom exception for this

**Comment by pragyanmehrotra** [comment]
Hi, @rafaeelaudibert I have incorporated the changes you have asked for and added a few test cases as well, Please have a look at them and lemme know if any further changes are needed.

**Comment by rafaeelaudibert** [comment]
Hey @pragyanmehrotra , just passing by to see if you have any updates on that?

**Comment by pragyanmehrotra** [comment]
Hi, @rafaeelaudibert Sorry for the late reply and a bit of inactivity I had a lot of quizzes this week and was a little busy with deadlines I'll drop the changes by this weekend.

**Comment by rafaeelaudibert** [comment]
Hey @pragyanmehrotra, don't worry! Be free to take the time you need to take for it! Good luck with the deadlines.

**Comment by pragyanmehrotra** [comment]
Yeah got a heavy weekend, had to implement the whole GUI part and the animations of a Game as a part of the deadline due Sunday night, Fortunately was able to get something done in time. Have a look at the tests I have added for the api and lemme know if some further changes are needed

**Comment by rafaeelaudibert** [comment]
That's awesome! Thank you very much @pragyanmehrotra, have a good end of the year!

**Pull request merged**